### ISSUE TITLE
feat: Split Schedules based on Environment

### ISSUE THREAD

**Issue by devon-seitz-cirrusmd** [meltano/meltano#6853]
### Feature scope

Configuration (settings parsing, validation, etc.)

### Description

It would be helpful to be able to have schedules split by environment. We have a use case where some schedules are run in staging, while other schedules are run in production. The current state is that all schedules (both staging & production) are loaded into both environments, but we do not turn on the production jobs in staging and vice versa for staging jobs.  

The outcome of this would be clearer differentiation between what should be running in the respective environments and reduced human error in the use of the tool (someone accidentally turning on a staging job in production for example).

May be resolved by:

- https://github.com/meltano/airflow-ext/pull/36

**Comment by aaronsteers** [comment]
Hi, @devon-seitz-cirrusmd - we did discuss this and are interested in this feature. 

For the use case you have in mind, what do you think of a plural `environments:` array being added as a list of environments to run against, versus a singular `environment:` declaration that pairs the schedule to one and only one environment?

Note, for test purposes, users could presumably force an execution using `meltano schedule run`, but the declaration would make the orchestration and scheduling features have a strong affinity to only the environment(s) named in the declaration.

**Comment by devon-seitz-cirrusmd** [comment]
@aaronsteers just so I make sure I am on the same page, it would be something like the below? In the first example the schedule would be active in both environments, and in the second example only in staging? If so that is a much better idea and I love it!

`
schedules:
  - name: example1
    interval: "@hourly"
    extractor: tap-postgres--example1
    loader: target-snowflake--example1
    transform: skip
    start_date: 2022-06-14 18:35:29.947934
    environments: [staging, production]
`

`
schedules:
  - name: example2
    interval: "@hourly"
    extractor: tap-postgres--example2
    loader: target-snowflake--example2
    transform: skip
    start_date: 2022-06-14 18:35:29.947934
    environments: [staging]
`

**Comment by aaronsteers** [comment]
Yep, exactly that! Optionally you could use this syntax but yaml tests both the same so there'd be no functional difference between the two declarations.

>     environments:
>     - staging
>     - production

Presumably, we'd have to update the airflow logic as well to respect this new specification. 

To keep this from being a breaking change for users, we could start with a rule: if any schedules declare their environments, then all should. That way revising users would continue to get their current behaviors until they start to move to explicit environment declarations.

What do you think?

**Comment by aaronsteers** [comment]
Speaking with @DouweM, he suggested (and I agree) that it could be cleaner to declare environment-specific schedules under the environment declaration.

There are a few challenges with this approach though. As of today, Meltano does not let you split an environment definition across files, so then all schedules relevant to the environment would need to be in the same yaml file as other environment properties.

- #6727

Additionally, and similar to the above logic, we probably need to assert that if _any_ schedules are defined at the environment level, then all should be. With this logic in place, an environment could override base schedules and simply declare `schedules: []` as an empty set, which would disable all schedules from running against the environment.
 
Most likely the effect of allowing environment-specific schedules is that once a use adopts the new convention, they'd like remove the base `schedules:` declaration and move all schedules under the appropriate environment definition.

**Comment by aaronsteers** [comment]
WIP here:

- https://github.com/meltano/airflow-ext/pull/36

**Comment by WillDaSilva** [comment]
Still relevant
